pennylane/templates/subroutines/single_excitation_unitary.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``SingleExcitationUnitary`` template.
+"""
+from pennylane import numpy as np
+
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.ops import CNOT, RX, RZ, Hadamard
+from pennylane.templates.decorator import template
+from pennylane.templates.utils import (
+    check_no_variable,
+    check_shape,
+    check_type,
+    check_wires,
+    get_shape,
+)
+
+
+@template
+def SingleExcitationUnitary(weight, wires=None):
```

Comment:
General question about PennyLane and templates: would it be helpful to add example usage to the docstrings? It's something we've been doing in SF
@mariaschuld

Reply:
Yes for sure. Here is the general template we are recommending in PL:

```
Single sentence that summarizes the function.

Multi-line description of the function (optional, if required).

Args:
    arg1 (type): description
    arg2 (type): description

Returns:
    type: description

**Example**

Minimal example with 1 or 2 code blocks (required)

.. UsageDetails::

    (optional) More complicated use cases, options, and larger code blocks

**Tutorials using <function>**

Cards linking to any relevant PennyLane demos (optional)
```